lib/features/mailbox/presentation/mailbox_controller.dart

```diff
@@ -281,31 +272,29 @@ class MailboxController extends BaseMailboxController with MailboxActionHandlerM
   }
 
   Future<void> refreshAllMailbox() async {
-    final session = mailboxDashBoardController.sessionCurrent;
-    final accountId = mailboxDashBoardController.accountId.value;
     if (session != null && accountId != null) {
-      consumeState(getAllMailboxInteractor!.execute(session, accountId));
+      consumeState(getAllMailboxInteractor!.execute(session!, accountId!));
     } else {
       consumeState(Stream.value(Left(GetAllMailboxFailure(NotFoundSessionException()))));
     }
   }
   
-  void _refreshMailboxChanges({jmap.State? currentMailboxState, Properties? properties}) {
-    log('MailboxController::_refreshMailboxChanges(): currentMailboxState: $currentMailboxState');
-    final newMailboxState = currentMailboxState ?? this.currentMailboxState;
-    log('MailboxController::_refreshMailboxChanges(): newMailboxState: $newMailboxState');
-    final accountId = mailboxDashBoardController.accountId.value;
-    final session = mailboxDashBoardController.sessionCurrent;
-    if (accountId != null && session != null && newMailboxState != null) {
-      refreshMailboxChanges(
-        session,
-        accountId,
-        newMailboxState,
-        properties: properties
-      );
-    } else {
+  void _refreshMailboxChanges({jmap.State? newState}) {
+    log('MailboxController::_refreshMailboxChanges():newState: $newState');
+    if (accountId == null ||
+        session == null ||
+        currentMailboxState == null ||
+        newState == currentMailboxState) {
```

Comment:
Imagine the following

User  update A, state is newStateA
User  update B, state is newStateB
User receive WS notif for newStateA -> resynch
User receive WS notif for newStateB -> resynch

If we maintain a cache (128 entries FIFO) that we could easily discard both notifications as they are for items WE generated.

Reply:
As for the case you gave I don't think it misses data in either case. We just need to have the old state with any new state, we can always synchronize it.

Reply:
Hope I get your point. You want to reduce the `resynch` with the state generated by the current client device?

Reply:
> No please. It's 7pm in VN Go sleep, have time with wife and kid. Tomorrow think back about it. Re-read carefully the example I gave. Run it on-paper to understand why it resynced twice yet it was uneeded. And if you need help throughtout this process we can set up a meeting where I explain things.

Oke. We will use queue to handle state when receive multiple state from websocket.